src/types.rs

```diff
@@ -247,6 +247,32 @@ pub struct TransactionDetails {
     pub confirmation_time: Option<BlockTime>,
 }
 
+impl PartialOrd for TransactionDetails {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for TransactionDetails {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        // Sort by timestamp, but if there's no timestamp put it first,
+        // if timestamps are equal, compare by txid
+        match (&self.confirmation_time, &other.confirmation_time) {
+            (Some(a), Some(b)) => {
+                let cmp = a.cmp(&b);
```

Comment:
I think you can make this (and below) quite a bit more concise by chaining the `cmp`s via [`then_with`](https://doc.rust-lang.org/std/cmp/enum.Ordering.html#method.then_with). 

Btw., `Ord` is implemented for `Option<T>`, so you should be able to just do `self.confirmation_time.cmp(&other.confirmation_time).then_with(...)`, no need for the `match`.